frontend/src/Dashboard/HistoricData.js

```diff
@@ -41,10 +48,40 @@ export default function HistoricData() {
     fetchData();
   }, [id]);
 
+  useEffect(() => {
+    async function fetchData() {
+      if (station.length > 0) {
+        try {
+          const response = await axios.get(`/api/latest?stationId=${station}`);
+          response.data && setWeatherData(response.data);
```

Comment:
Ez nem annyira elegáns, olvashatóbb, ha egy if-et írsz ebben az esetben.

Reply:
ezt most nem értem :D @6uliver

Reply:
a setWeatherData függvényhívás azért fut le, mert a response.data igazra értékelődik ki, így az && második argumentumát is ki kell értékelni, ami pont egy függvényhívás. Logikailag működik, csak nem olvasmányos. Sokkal egyértelműbben olvasható és módosítható lenne, ha nem egy JS-es kifejezés mellékhatásaként hívnánk meg a függvényt, hanem egy if szerkezetben explicitebben:

```
if (response.data) {
    setWeatherData(response.data);
}
```